File: deck/seed/code/native/node/runtime/variable.ts
// Environment variable runtime for node. Reached only through the public environment API.
const variable = {
  get: (name: string): string => process.env[name] ?? '',
  set: (name: string, value: string): void => {
    process.env[name] = value
  },
  remove: (name: string): void => {
    delete process.env[name]
  },
  list: (): Record<string, string> =>
    Object.fromEntries(
      Object.entries(process.env).filter(
        (entry): entry is [string, string] => entry[1] !== undefined,
      ),
    ),
  check: (name: string): boolean => process.env[name] !== undefined,
}

File: deck/seed/code/native/swift/runtime/variable.swift
// Environment variable runtime. Reached only through the public environment API.
import Foundation

enum variable {
    static func get(_ name: String) -> String {
        ProcessInfo.processInfo.environment[name] ?? ""
    }

    static func set(_ name: String, _ value: String) {
        setenv(name, value, 1)
    }

    static func remove(_ name: String) {
        unsetenv(name)
    }

    static func list() -> SeedMap<String, String> {
        SeedMap(ProcessInfo.processInfo.environment)
    }

    static func check(_ name: String) -> Bool {
        ProcessInfo.processInfo.environment[name] != nil
    }
}
